packages/components/src/filter-picker/index.js

```diff
@@ -143,7 +167,35 @@ class FilterPicker extends Component {
 		config.staticParams.forEach( ( param ) => {
 			update[ param ] = query[ param ];
 		} );
-		updateQueryString( update, path, persistedQuery );
+
+		// Remove all of this filter's params not associated witth the update while
+		// leaving any other params from any other filter an extension may have added.
+		this.getAllFilterParams().forEach( ( param ) => {
```

Comment:
Probably premature optimization, but `getAllFilterParams()` only needs to be called once for every prop update.

Reply:
This function call to `getAllFilterParams` is executed in the `update` function which is only called when the user clicks selects a filter.